src/Files.App/Utils/Shell/Win32API.cs

```diff
@@ -231,22 +286,17 @@ public static string ExtractStringFromDLL(string file, int number)
 		/// <param name="thumbnailSize"></param>
 		/// <param name="isFolder"></param>
 		/// <param name="getIconOnly"></param>
-		/// <param name="getOverlay"></param>
-		/// <param name="onlyGetOverlay"></param>
 		/// <returns></returns>
-		public static (byte[]? icon, byte[]? overlay, bool isIconCached) GetFileIconAndOverlay(
+		public static (byte[]? icon, bool isIconCached) GetIcon(
```

Comment:
You _could_ use out variable for isIconCached.

Reply:
What changes would need to be made and what's the benefit?

Reply:
Just to avoid returning taple type. I'd like methods to return single value(struct type, pre-defined types) to have single resposibility.

Now
```cs
public static async Task<(byte[] IconData, bool isIconCached)> GetIconAsync(
	string path,
	uint requestedSize,
	bool isFolder,
	bool getThumbnailOnly,
	IconOptions iconOptions)
```

After
```cs
public static async Task<byte[] IconData> GetIconAsync(
	string path,
	uint requestedSize,
	bool isFolder,
	bool getThumbnailOnly,
	IconOptions iconOptions,
	out bool isIconCached)
```

Besides, I see isIconCached often is not used.
So when you want to discard, you can just do:

```cs
public static async Task<byte[] IconData> GetIconAsync(
	"",
	0,
	false,
	false,
	IconOptions.None,
	out var _)
```

Reply:
But this is just my preferences for codebase quality.

Reply:
We can do this in a future phase 👍